Exercise_5.java

```diff
@@ -1,36 +1,58 @@
-class IterativeQuickSort { 
-    void swap(int arr[], int i, int j) 
-    { 
-	//Try swapping without extra variable 
-    } 
-  
-    /* This function is same in both iterative and 
-       recursive*/
-    int partition(int arr[], int l, int h) 
-    { 
-        //Compare elements and swap.
-    } 
-  
-    // Sorts arr[l..h] using iterative QuickSort 
-    void QuickSort(int arr[], int l, int h) 
-    { 
-        //Try using Stack Data Structure to remove recursion.
-    } 
-  
-    // A utility function to print contents of arr 
-    void printArr(int arr[], int n) 
-    { 
-        int i; 
-        for (i = 0; i < n; ++i) 
-            System.out.print(arr[i] + " "); 
-    } 
-  
-    // Driver code to test above 
-    public static void main(String args[]) 
-    { 
-        IterativeQuickSort ob = new IterativeQuickSort(); 
-        int arr[] = { 4, 3, 5, 2, 1, 3, 2, 3 }; 
-        ob.QuickSort(arr, 0, arr.length - 1); 
-        ob.printArr(arr, arr.length); 
-    } 
-} 
\ No newline at end of file
+// Java program for implementation of QuickSort
+import java.util.*;
+
+class IterativeQuickSort {
+	
+	static int partition(int arr[], int low, int high)
+	{
+		int pivot = arr[high];
+		int i = (low - 1); // index of smaller element
+		for (int j = low; j <= high - 1; j++) {
+			// If current element is smaller than or
+			// equal to pivot
+			if (arr[j] <= pivot) {
+				i++;
+
+				// swap arr[i] and arr[j]
+				int temp = arr[i];
+				arr[i] = arr[j];
+				arr[j] = temp;
+			}
+		}
+
+		// swap arr[i+1] and arr[high] (or pivot)
+		int temp = arr[i + 1];
+		arr[i + 1] = arr[high];
+		arr[high] = temp;
+
+		return i + 1;
+	}
+
+	static void qSort(int arr[], int low, int high)
+	{
+		if (low < high) {
+			/* pi is partitioning index, arr[pi] is
+			now at right place */
+			int pi = partition(arr, low, high);
+
+			// Recursively sort elements before

```

Comment:
Instead of using recursion, you have to implement quick sort in iterative manner. You will be using stack here.